File: android/app/src/main/java/com/onchainwalletdemo/modules/WalletSettingsModule.kt
package com.onchainwalletdemo.modules

import android.content.Context
import android.content.SharedPreferences
import expo.modules.core.ExportedModule
import expo.modules.core.Promise
import expo.modules.core.arguments.ReadableArguments
import expo.modules.core.interfaces.ExpoMethod

class WalletSettingsModule(context: Context) : ExportedModule(context) {
    private val sharedPreferences: SharedPreferences = context.getSharedPreferences("WalletSettings", Context.MODE_PRIVATE)

    override fun getName(): String {
        return "WalletSettingsModule"
    }

    @ExpoMethod
    fun toggleCurrency(promise: Promise) {
        val currentCurrency = sharedPreferences.getString("fiatCurrency", "USD") ?: "USD"
        val newCurrency = if (currentCurrency == "USD") "HKD" else "USD"
        
        sharedPreferences.edit().putString("fiatCurrency", newCurrency).apply()
        
        // 发送事件到JS
        val map = mapOf("currency" to newCurrency)
        sendEvent("onCurrencyChanged", map)
        
        promise.resolve(newCurrency)
    }

    @ExpoMethod
    fun getCurrentCurrency(promise: Promise) {
        val currentCurrency = sharedPreferences.getString("fiatCurrency", "USD") ?: "USD"
        promise.resolve(currentCurrency)
    }

    private fun sendEvent(eventName: String, params: Map<String, Any>) {
        // 在实际应用中，你需要使用合适的事件发射器
        // 这里仅作为示例，实际实现可能需要根据Expo模块API进行修改
    }
} 

File: ios/WalletSettingsModule/WalletSettingsModule.swift
import Foundation
import ExpoModulesCore

public class WalletSettingsModule: Module {
  // 定义事件发送器
  public let currencyChangedEvent = EventEmitter()
  
  // 导出方法给JS调用
  @objc public func toggleCurrency(_ promise: Promise) {
    // 在iOS端获取当前的货币设置
    let defaults = UserDefaults.standard
    let currentCurrency = defaults.string(forKey: "fiatCurrency") ?? "USD"
    
    // 切换货币类型
    let newCurrency = currentCurrency == "USD" ? "HKD" : "USD"
    defaults.set(newCurrency, forKey: "fiatCurrency")
    
    // 发送事件通知JS端
    currencyChangedEvent([
      "currency": newCurrency
    ])
    
    // 返回新的货币类型
    promise.resolve(newCurrency)
  }
  
  // 获取当前货币设置
  @objc public func getCurrentCurrency(_ promise: Promise) {
    let defaults = UserDefaults.standard
    let currentCurrency = defaults.string(forKey: "fiatCurrency") ?? "USD"
    promise.resolve(currentCurrency)
  }
  
  // 导出给JS的模块定义
  public func definition() -> ModuleDefinition {
    Name("WalletSettingsModule")
    
    Events("onCurrencyChanged")
    
    Function("toggleCurrency", toggleCurrency)
    Function("getCurrentCurrency", getCurrentCurrency)
  }
} 